apps/api/src/workspace/controller/workspace.controller.ts

```diff
@@ -220,6 +220,25 @@ export class WorkspaceController {
     @Param('workspaceId') workspaceId: Workspace['id'],
     @Param('searchTerm') searchTerm: string
   ) {
-    return this.workspaceService.globalSearch(workspaceId, searchTerm, user);
+    // Check permissions before proceeding
+    if (
+      !user ||
+      !(await this.workspaceService.hasAuthorities(
```

Comment:
You won't need to do this check. This will be automatically done. Just include the authorities using `@RequiredAuthorities` decorator. Refer this:
https://github.com/keyshade-xyz/keyshade/blob/56bda0d8eedea534ef12887cad103a8b5933babd/apps/api/src/project/controller/project.controller.ts#L72